### PR TITLE
Move hive-scratch to user-directories

### PR DESCRIPTION
**Pull request by aespinosa**
So that hive sessions are under quota

Need to backport to master

### CODE REVIEW

File: cookbooks/bcpc-hadoop/recipes/hive_hcatalog.rb [hunk 1]
```diff
@@ -53,15 +53,6 @@
...
 end
 
-bash 'create-hive-scratch' do
-  code <<-EOH
-  hdfs dfs -mkdir -p #{node['bcpc']['hadoop']['hive']['scratch']['dir']}
-  hdfs dfs -chmod 1777 #{node['bcpc']['hadoop']['hive']['scratch']['dir']}
-  hdfs dfs -chown hive:hdfs #{node['bcpc']['hadoop']['hive']['scratch']['dir']}
-  EOH
-  user 'hdfs'
-end
```
Comment (vt0r) [inline cookbooks/bcpc-hadoop/recipes/hive_hcatalog.rb]: Pretty sure this block is still required. We want to ensure the directory exists and is owned/permed properly. Obviously we'll have to replace `1777` with something more sane like `0700` and `hive:hdfs` with the proper user/group combo we use for user dirs.

**Comment by aespinosa** [comment]
(Needs a cluster build)

File: cookbooks/bcpc-hadoop/recipes/hive_hcatalog.rb (hunk 1 above)
Comment (cbaenziger) [inline cookbooks/bcpc-hadoop/recipes/hive_hcatalog.rb]: We shouldn't need hive to own something under the user's directory right @vt0r ?

Comment (vt0r) [inline cookbooks/bcpc-hadoop/recipes/hive_hcatalog.rb]: No, sir. We'd want to fill in the chown user dynamically to match the user it's being created for. Probably want `0700` perms as well

Comment (cbaenziger) [inline cookbooks/bcpc-hadoop/recipes/hive_hcatalog.rb]: Allan, can you check if Hive will create the directory if the user has perms and it doesn't exist? That way we can be lazy and blow this off (and nice not to run `O (users)` more HDFS actions).

**Comment by aespinosa** [comment]
Rescope to 3.1

**Pull request closed without merging**